mysystehis/main.tex

```diff
@@ -0,0 +1,1155 @@
+ 	  Input.GetAxis("Vertical"));
+ 	 \end{lstlisting}
+ 	 Next, i have to normalize the input vector to see which direction the player have to move , face correctly. To be more specify, let's consider two trigonometry below:
+ 	
+ 	 		\definecolor{qqwuqq}{rgb}{0.,0.39215686274509803,0.}
+ 	 		\definecolor{ffqqqq}{rgb}{1.,0.,0.}
+ 	 		\definecolor{uuuuuu}{rgb}{0.26666666666666666,0.26666666666666666,0.26666666666666666}
+ 	 		\begin{tikzpicture}[line cap=round,line join=round,>=triangle 45,x=0.5cm,y=0.5cm]
+ 	 		\clip(-17.62,-11.46) rectangle (13.08,6.28);
+ 	 		\draw [shift={(-10.3,-3.27)},line width=0.5pt,color=qqwuqq,fill=qqwuqq,fill opacity=0.10000000149011612] (0,0) -- (0.4233601239113539:0.6) arc (0.4233601239113539:49.49045192111633:0.6) -- cycle;
+ 	 		\draw [line width=0.5pt] (-14.36,-3.3)-- (-6.24,-3.24);
+ 	 		\draw [line width=0.5pt] (-10.3,-3.27) circle (2.030055417962771cm);
+ 	 		\draw [line width=0.5pt] (-10.3,-3.27)-- (-6.24,-3.24);
+ 	 		\draw [line width=0.5pt] (-10.33,0.79)-- (-10.27,-7.33);
+ 	 		\draw [line width=2.8pt,color=ffqqqq] (-10.3,-3.27)-- (-7.662654488417639,-0.1831069580342808);
+ 	 		\draw [line width=0.5pt] (-7.662654488417639,-0.1831069580342808)-- (-7.639990169721227,-3.2503447549486797);
+ 	 		\draw (-10.74,1.68) node[anchor=north west] {90};
+ 	 		\draw [line width=0.5pt] (1.77,-3.3) circle (2.030055417962771cm);
+ 	 		\draw [line width=0.5pt] (-2.25,-3.31)-- (5.87,-3.25);
+ 	 		\draw (-5.82,-2.98) node[anchor=north west] {0};
+ 	 		\draw (-15.86,-3.14) node[anchor=north west] {180};
+ 	 		\draw (-10.94,-7.76) node[anchor=north west] {270};
+ 	 		\draw (1.46,1.66) node[anchor=north west] {0};
+ 	 		\draw (6.42,-2.92) node[anchor=north west] {90};
+ 	 		\draw (1.28,-7.96) node[anchor=north west] {180};
+ 	 		\draw (-3.74,-3.02) node[anchor=north west] {270};
+ 	 		\draw [line width=2.8pt,color=ffqqqq] (1.77,-3.3)-- (1.7404368698694395,0.7600032045968739);
+ 	 		\draw [line width=0.5pt] (1.77,-3.3)-- (1.7995631301305606,-7.3600032045968735);
+ 	 		\begin{scriptsize}
+ 	 		\draw [fill=uuuuuu] (-10.3,-3.27) circle (2.0pt);
+ 	 		\draw[color=black] (-8.94,-3.55) node {$X$};
+ 	 		\draw[color=black] (-7.32,-1.87) node {$Y$};
+ 	 		\draw[color=qqwuqq] (-9.22,-2.87) node {$\alpha$};
+ 	 		\draw [fill=uuuuuu] (1.77,-3.3) circle (2.0pt);
+ 	 		\end{scriptsize}
+ 	 		\end{tikzpicture}
+ 	 	\\[0.02cm]
+ 	 	From the left one above, suppose that the red line is the input direction, X and Y are vertical and horizontal component. Now, i have to find the angle {$\alpha$} which is the direction the character will rotate, calculated as: 
+ 	 	\\[-0.5cm]
+ 	 	\[\alpha = arctan(\frac{Y}{X})\]
+ 	 	In Unity, however, if my character is facing forward, then it has the rotation of 0 (as seen from the right one), when facing right it has the rotation of 90 degree and so on. Let's call the rotation of character in unity is r, when looking at the two circle, it's clearly that r is calculated as: \\[-0.5cm]
```

Comment:
Let's call the rotation of the character "r". When looking at the two circles, it is clear that r is calculated as: